spring-cloud-task-core/src/test/java/org/springframework/cloud/task/configuration/RepositoryTransactionManagerConfigurationTests.java

```diff
@@ -0,0 +1,237 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.test.jdbc.JdbcTestUtils;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.DefaultTransactionStatus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Michael Minella
+ */
+public class RepositoryTransactionManagerConfigurationTests {
+
+	@Test
+	public void testZeroCustomTransactionManagerConfiguration() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+			.withConfiguration(
+				AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+					SimpleTaskAutoConfiguration.class,
+					ZeroTransactionManagerConfiguration.class))
+			.withPropertyValues("application.name=transactionManagerTask");
+
+		applicationContextRunner.run((context) -> {
+			DataSource dataSource = context.getBean("dataSource", DataSource.class);
+
+			int taskExecutionCount = JdbcTestUtils
+				.countRowsInTable(new JdbcTemplate(dataSource), "TASK_EXECUTION");
+
+			assertThat(taskExecutionCount).isEqualTo(1);
+		});
+	}
+
+	@Test
+	public void testSingleCustomTransactionManagerConfiguration() {
+		testConfiguration(SingleTransactionManagerConfiguration.class);
+	}
+
+	@Test
+	public void testMultipleCustomTransactionManagerConfiguration() {
+		testConfiguration(MultipleTransactionManagerConfiguration.class);
+	}
+
+	private void testConfiguration(Class configurationClass) {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+			.withConfiguration(
+				AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+					SimpleTaskAutoConfiguration.class,
+					configurationClass))
+			.withPropertyValues("application.name=transactionManagerTask");
+
+		applicationContextRunner.run((context) -> {
+			DataSource dataSource = context.getBean("dataSource", DataSource.class);
+
+			int taskExecutionCount = JdbcTestUtils
+				.countRowsInTable(new JdbcTemplate(dataSource), "TASK_EXECUTION");
+
+			// Verify that the create call was rolled back
+			assertThat(taskExecutionCount).isEqualTo(0);
+
+			// Execute a new create call so that things close cleanly
+			TaskRepository taskRepository = context.getBean("taskRepository",
+				TaskRepository.class);
+
+			TaskExecution taskExecution = taskRepository
+				.createTaskExecution("transactionManagerTask");
+			taskExecution = taskRepository.startTaskExecution(2, "transactionManagerTask",
+				new Date(), new ArrayList<>(0), null);
+
+			TaskLifecycleListener listener = context.getBean(TaskLifecycleListener.class);
+
+			ReflectionTestUtils.setField(listener, "taskExecution", taskExecution);
```

Comment:
We need to check that the taskExecutionCount == 1.

Reply:
We can, but it wouldn't have anything to do with this test if it wasn't.  The test is to verify that we are using the correct transaction manager.  We prove that in the first assertion.  Everything after the assertion is just to allow the context to close gracefully and not needed to validate the functionality.

Reply:
In this case it would make it consistent with the tests in `testZeroCustomTransactionManagerConfiguration`.

Reply:
`testZeroCustomTransactionManagerConfiguration` is there just to make sure I didn't break anything and that the out of the box functionality still works.  The two other tests only need one assertion to confirm the functionality.  What I mean by that is, if you add your additional assertion and then change my code change back, that assertion doesn't get fired.  Nothing related to my code change is validated by that additional assertion.